### ISSUE TITLE
PaginatedIterator has duplicate call

### ISSUE THREAD

**Issue by Katsute** [Katsute/MyAnimeList-Java-API#28]
`hasNextPage` is called twice for some reason

https://github.com/Katsute/MyAnimeList-Java-API/blob/270092b5596e0b08f7a8c787341ffa33fe592048/src/main/java/com/kttdevelopment/myanimelist/PaginatedIterator.java#L23-L39

**Comment by Katsute** [comment]
Issue is caused by how for loops use iterators, first they check hasNext() to see if there is a page then calls the next(). 
This calls the nextPage first when checking if there is a next item then when they actually request the next item.

This needs to be changed so the nextPage is instead a Boolean that is set when the current page is requested and changes only when the page changes (this would also reduce the amount of requests from 3 to 1 since the initial request already has information on the next page.

tldr. save next page bool when requesting the current page instead of a new http call

**Comment by Katsute** [comment]
Optimizing this is actually more difficult than it appears. This optimization will not be fixed in this version.

**Comment by Katsute** [comment]
Fixed in 71838e53766ff7c95b74af3dcf12d332f6b59867

We can only lower this to two requests. One request wouldn't work since the search returns a list and not an object.

**Comment by Katsute** [comment]
This can be further optimized to one request by copying the search method 
https://github.com/Katsute/MyAnimeList-Java-API/blob/ef8fafcf4197d740cbe3503ee50300c6b3f400cf/src/main/java/com/kttdevelopment/myanimelist/MyAnimeListImpl.java#L70-L87
to here
https://github.com/Katsute/MyAnimeList-Java-API/blob/ef8fafcf4197d740cbe3503ee50300c6b3f400cf/src/main/java/com/kttdevelopment/myanimelist/MyAnimeListImpl.java#L116-L126
so we then can easily check if paging contains a next page and return that fixed bool instead of initiating a request.

**Comment by Katsute** [comment]
Needs to be changed b/c it fails page. Have next page read the link with regex to determine the next offset.
